### ISSUE TITLE
Misleading documentation on externals type or potential bug

### ISSUE THREAD

**Issue by dmitrysteblyuk** [webpack/webpack#11960]
The [documentation](https://webpack.js.org/configuration/externals/#string) makes us believe that this example:
```
module.exports = {
  //...
  externals: {
    jquery: 'jQuery'
  }
};
```

will result in `jquery` imports to be replaced with `jQuery` variable:
```
In other words, when a string is provided it will be treated as root (defined above and below).
```
```
root: The library should be available as a global variable (e.g. via a script tag).
```

However it's not exactly true.
The default type of externals is defined by `externalsType` option. Which is by default defined as follows:

https://github.com/webpack/webpack/blob/master/lib/config/defaults.js#L189-L198

So for example in case of commonjs library target `import $ from 'jquery';` will be transformed to `require("jQuery")`.

**Comment by sokra** [comment]
Yes the documentation may need some improvements here. Good that you discovered that. Do you want to send a PR to improve the documentation?

**Comment by dmitrysteblyuk** [comment]
So it's the expected behaviour then?
I guess then external type ideally should always be set explicitly like `jquery: 'var jQuery'` (so that it works no matter which library target is selected later on).

> Do you want to send a PR

sure, I'll see what I can do.
